source/plugins/go/input/lib/proxyutils.go

```diff
@@ -0,0 +1,91 @@
+package lib
+
+import (
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+var proxyCertPath = "/etc/ama-logs-secret/PROXYCERT.crt"
+
+type ProxyConfiguration struct {
+	User string
+	Pass string
+	Addr string
+	Port string
+}
+
+// func getProxyConfiguration() map[string]string {
+// 	amalogsproxySecretPath := "/etc/ama-logs-secret/PROXY"
+// 	if _, err := os.Stat(amalogsproxySecretPath); os.IsNotExist(err) {
+// 		return make(map[string]string)
+// 	}
+
+// 	proxyConfig, err := parseProxyConfiguration(amalogsproxySecretPath)
+// 	if err != nil {
+// 		// todo: handle error
+// 		return make(map[string]string)
+// 	}
+
+// 	// todo: log if empty
+// 	return proxyConfig
+// }
+
+func getProxyEndpoint() string {
+	amaLogsProxySecretPath := "/etc/ama-logs-secret/PROXY"
+	proxyConfig, err := ioutil.ReadFile(amaLogsProxySecretPath)
```

Comment:
File /etc/ama-logs-secret/PROXY only exists if the customer configured http proxy, so we would need to check the existence of the file. If the file doesnt exist,  return "" and else if the file exists but fail to read, then throw error. Please double check all the cases of existing fluentd  scenarios will be captured in the Go code.

Reply:
in ruby, for all scenarios we pass {} if we hit any hiccups. I followed the same pattern here. refer: https://github.com/microsoft/Docker-Provider/blob/68544733aa28eb86bb8b0bcd3a1ef5801e330b57/source/plugins/ruby/proxy_utils.rb#L7